app/components/Navbar/SavedItems.tsx

```diff
@@ -1,10 +1,96 @@
-import { Button } from "@/app/components/ui/button";
-import { Icons } from "@/app/utils/icons";
+"use client";
+import Image from "next/image";
+
+import Link from "next/link";
+import { cn } from "@/app/utils";
+import { ConditionEnum } from "@/types";
+
+import DeleteSavedProducts from "./delete-saved-products";
+import { renderConditionIcon } from "../ProductPage/ProductDetails";
+import { useProductStore } from "@/hooks/productStore";
+
+export interface SavedProduct {
+  _id: string;
+  Image: string;
+  Condition: keyof typeof ConditionEnum;
+  Price: number;
+  Negotiable: boolean;
+  Product_Name: string;
+}
+
+export default function SavedItems() {
+  const { savedProducts } = useProductStore();
 
-export function SavedItems() {
   return (
-    <Button variant="ghost" size="icon">
-      <Icons.saved className="h-[1.4rem] w-[1.4rem] " />
-    </Button>
+    <aside className="w-full px-4 sm:max-w-sm md:max-w-md">
+      <header className="mb-6 mt-5">
+        <h1 className="text-2xl font-bold">Saved Products</h1>
+        <p className="text-muted-foreground">Manage your saved products</p>
+      </header>
+      <ul
+        className={cn(
+          "h-[calc(100dvh-7rem)] overflow-y-auto",
+          "flex flex-col space-y-4",
+        )}
+      >
+        {Array.from(savedProducts?.entries() ?? []).map(([id, product]) => (
```

Comment:
```suggestion
        {(...savedProducts?.entries() ?? []).map((id, product) => (
```

Reply:
but its the same thing. then what changes by doing that?

Reply:
Compiler